desc/compute/utils.py

```diff
@@ -140,6 +140,16 @@ def _compute(
         data = data_index[parameterization][name]["fun"](
             params=params, transforms=transforms, profiles=profiles, data=data, **kwargs
         )
+        # to reduce repeated code, for 3D and naturally in rtz but desired in xyz basis
+        # quantities can be converted to xyz basis here
+        if (
```

Comment:
move this outside of recursive, have _compute functions assume things come in in rpz